widget/toolbar.go

```diff
@@ -13,31 +13,48 @@ type ToolbarItem interface {
 	ToolbarObject() fyne.CanvasObject
 }
 
+type toolbarActionButton struct {
+	Button
+
+	t *ToolbarAction
+}
+
+func (b *toolbarActionButton) Refresh() {
+	b.Icon = b.t.Icon
+	b.OnTapped = b.t.OnActivated
+
+	if b.t.Disabled {
+		b.Button.Disable()
```

Comment:
Note that this calls Refresh implicitly, so I avoided calling it explicitly again after.